mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/Attachment.java

```diff
@@ -29,7 +33,7 @@ public Attachment(InputStream stream) {
         this.stream = stream;
     }
 
-    public InputStream getStream() {
-        return stream;
+    public InputStream getStream() throws IOException {
+        return new ByteArrayInputStream(IOUtils.toByteArray(stream));
```

Comment:
I think you still have an issue, as the stream field can be consumed only once.
Can you add a test about this?